data_model/src/executor.rs

```diff
@@ -47,10 +52,87 @@ mod model {
         pub wasm: WasmSmartContract,
     }
 
+    /// Executor data model.
+    ///
+    /// Defined from within the executor, it describes certain structures the executor
+    /// works with.
+    ///
+    /// Executor can define:
+    ///
+    /// - Permission tokens (see [`crate::permission::PermissionToken`])
+    /// - Configuration parameters (see [`crate::parameter::Parameter`])
+    #[derive(
+        Default,
+        Debug,
+        Clone,
+        PartialEq,
+        Eq,
+        PartialOrd,
+        Ord,
+        Constructor,
+        Decode,
+        Encode,
+        Deserialize,
+        Serialize,
+        IntoSchema,
+        Display,
+    )]
+    #[ffi_type]
+    #[display(fmt = "{self:?}")]
+    pub struct ExecutorDataModel {
+        /// Data model JSON schema, typically produced by [`IntoSchema`].
+        pub schema: JsonString,
+        /// Permission tokens supported by the executor.
+        ///
+        /// These IDs refer to the types in the schema.
+        pub permission_token_ids: BTreeSet<PermissionTokenId>,
+        /// Configuration parameters supported by the executor.
+        ///
+        /// These IDs refer to the types in the schema.
+        pub parameter_ids: BTreeSet<ParameterId>,
```

Comment:
```suggestion
        pub permission_tokens: BTreeSet<PermissionTokenId>,
        /// Configuration parameters supported by the executor.
        ///
        /// These IDs refer to the types in the schema.
        pub parameters: BTreeSet<ParameterId>,
```

how about this? it's simpler and it's not really necessary to emphasize they are ids